### PR TITLE
[Docs] Add permission setup page for the clouds

### PR DESCRIPTION
**Pull request by Michaelvll**
This PR adds the instructions for set up the cloud accounts with the required permissions.

Future TODO:
- [ ] Add instruction for Azure.

Tested (run the relevant ones):

- [x] Any manual or new tests for this PR (please specify below)
  - rendered locally
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: docs/source/reference/cloud-permissions.rst
Comment (concretevitamin) [inline docs/source/reference/cloud-permissions.rst:1]: Move this to the bottom of USER GUIDES? We can reserve "References" for CLI/API/YAML refs.

File: docs/source/reference/cloud-permissions.rst [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+---
+
+.. _cloud-permissions-aws-user-creation:
+
+User Creation
+~~~~~~~~~~~~~
+
+AWS accounts can be attached with a policy that limits the permissions of the account. Follow these steps to create an AWS user with the minimum permissions required by SkyPilot:
+
+1. Open the IAM dashboard in the AWS console and click on the "Users" tab. Then, click "Add users" and enter the user's name.
```
Comment (concretevitamin) [inline docs/source/reference/cloud-permissions.rst:22]: For the whole doc: for referring to buttons that appear in cloud consoles, e.g., `"Users"`, can we bold it `**Users**`? Following a common doc styleguide.

File: docs/source/getting-started/installation.rst [hunk 2]
```diff
@@ -62,8 +64,11 @@ To get the **AWS access key** required by :code:`aws configure`, please go to th
   $ # Configure your AWS credentials
   $ aws configure
 
+To create a new AWS user with minimal permissions for SkyPilot, see :ref:`AWS User Creation <cloud-permissions-aws>`.
```
Comment (concretevitamin) [inline docs/source/getting-started/installation.rst:67]: Can we add something like **Optional** and/or "not needed to get started with SkyPilot" disclaimer?

I am worried that it sounds like required steps.

Perhaps put it below L69.

Same for GCP?

**Comment by concretevitamin** [comment]
We may want to merge this soon :)

**Comment by Michaelvll** [comment]
Sorry for missing the reviews @concretevitamin! Just updated the doc and fixed all the warnings for the doc building. Please take another look. Thanks!

File: docs/source/getting-started/installation.rst [hunk 3]
```diff
@@ -66,13 +66,12 @@ To get the **AWS access key** required by :code:`aws configure`, please go to th
   $ # Configure your AWS credentials
   $ aws configure
 
-To create a new AWS user with minimal permissions for SkyPilot, see :ref:`AWS User Creation <cloud-permissions-aws>`.
+Note: If you are using AWS IAM Identity Center (AWS SSO), you will need :code:`pip install awscli>=1.27.10`. See `here <https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sso.html>`__ for instructions on how to configure AWS SSO.
```
Comment (concretevitamin) [inline docs/source/getting-started/installation.rst:69]: Merge with master? https://skypilot.readthedocs.io/en/latest/getting-started/installation.html#amazon-web-services-aws

File: docs/source/index.rst [hunk 4]
```diff
@@ -103,6 +103,7 @@ Documentation
    reference/quota
    reference/logging
    reference/faq
+   reference/cloud-permissions
```
Comment (concretevitamin) [inline docs/source/index.rst:106]: Thinking we should probably add a new section with one of these names:
- Cloud Administration
- Cloud Administration and Usage
- Cloud Setup and Usage
- better name

?

And then we can put
- cloud-permissions: Minimal Cloud Permissions Setup
- cloud-administration: Using Cloud-specific Features
  - may want to rename the filename too 
- quota: Requesting Quota

under it. Note I put some renamed titles above. How do they read?

File: docs/source/reference/cloud-permissions.rst [hunk 5]
```diff
@@ -15,23 +20,25 @@ AWS
 .. _cloud-permissions-aws-user-creation:
 
 User Creation
-~~~~~~~~~~~~~
+~~~~~~~~~~~~~~
 
 AWS accounts can be attached with a policy that limits the permissions of the account. Follow these steps to create an AWS user with the minimum permissions required by SkyPilot:
 
-1. Open the IAM dashboard in the AWS console and click on the "Users" tab. Then, click "Add users" and enter the user's name.
+1. Open the `IAM dashboard <https://us-east-1.console.aws.amazon.com/iamv2/home#/home>_` in the AWS console and click on the **Users** tab. Then, click **Add users** and enter the user's name. Click **Next**.
```
Comment (concretevitamin) [inline docs/source/reference/cloud-permissions.rst:27]: url rendered

File: docs/source/reference/cloud-permissions.rst [hunk 6]
```diff
@@ -100,30 +107,36 @@ AWS accounts can be attached with a policy that limits the permissions of the ac
...
+4. Click **Next: Tags** and follow the instructions to finish creating the policy. You can give the policy a descriptive name, such as ``minimal-skypilot-policy``.
+5. Go back to the previous window and click on the refresh button, and you can now search for the policy you just created.
 
 .. image:: ../images/screenshots/aws/aws-add-policy.png
     :width: 80%
     :align: center
     :alt: AWS Add Policy
-7. [Optional] If you would like to have your users access the s3 bucket. You can additionally attach the S3 access, such as the "AmazonS3FullAccess" policy.
+
+6. [Optional] If you would like to have your users access the s3 bucket. You can additionally attach the S3 access, such as the "AmazonS3FullAccess" policy.
```
Comment (concretevitamin) [inline docs/source/reference/cloud-permissions.rst:119]:
```suggestion
6. [Optional] If you would like to have your users access S3 buckets: You can additionally attach S3 access, such as the "AmazonS3FullAccess" policy.
```

File: docs/source/reference/cloud-permissions.rst [hunk 7]
```diff
@@ -100,30 +107,36 @@ AWS accounts can be attached with a policy that limits the permissions of the ac
...
     :width: 80%
     :align: center
     :alt: AWS Add S3 Policy
 
-8. Click on "Next" and follow the instructions to create the user.
+7. Click on **Next** and follow the instructions to create the user.
 
 With the steps above you are almost ready to have the users in your organization to use SkyPilot with the minimal permissions.
 
 **One more thing** to do is to create a single service account "skypilot-v1" for all the users in your organization. There are two ways to accomplish this:
```
Comment (concretevitamin) [inline docs/source/reference/cloud-permissions.rst:130]:
```suggestion
**One more thing** to do is to create a single service account ``skypilot-v1`` for all users in your organization. There are two ways to accomplish this:
```

File: docs/source/reference/cloud-permissions.rst [hunk 8]
```diff
@@ -146,11 +159,16 @@ With the steps above you are almost ready to have the users in your organization
                 ],
                 "Resource": "arn:aws:iam::<account-ID-without-hyphens>:instance-profile/skypilot-v1"
             }
-2. Instead, you can create the "skypilot-v1" service account manually. The following section describe how to create the service account manually.
+
+2. Alternatively, you can create the "skypilot-v1" service account manually. The following section describe how to create the service account manually.
```
Comment (concretevitamin) [inline docs/source/reference/cloud-permissions.rst:163]:
```suggestion
2. Alternatively, you can create the ``skypilot-v1`` service account manually. The following section describes how to create the service account manually.
```

File: docs/source/reference/cloud-permissions.rst [hunk 9]
```diff
@@ -146,11 +159,16 @@ With the steps above you are almost ready to have the users in your organization
...
 
 
 Service Account Creation
-~~~~~~~~~~~~~~~~~~~~~~~~
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. note::
+
+    If you already have a service account called "skypilot-v1" in your AWS account, it is likely created by SkyPilot automatically, and you can skip this section.
```
Comment (concretevitamin) [inline docs/source/reference/cloud-permissions.rst:170]:
```suggestion
    If you already have a service account called ``skypilot-v1`` in your AWS account, it is likely created by SkyPilot automatically, and you can skip this section.
```

File: docs/source/reference/cloud-permissions.rst (hunk 7 above)
Comment (concretevitamin) [inline docs/source/reference/cloud-permissions.rst:130]: Should we replace "service account" with "IAM role" in this doc? This is the AWS terminology (i.e., I found the role in https://us-east-1.console.aws.amazon.com/iamv2/home?region=us-east-1#/roles).

**Review by concretevitamin — COMMENTED** [review]
This is great @Michaelvll! Some notes.

File: docs/source/index.rst (hunk 4 above)
Comment (Michaelvll) [reply docs/source/index.rst]: Good point. I also re-ordered the "User Guides" a bit to de-emphasize the local mode. I also chose to use shorter name to make sure they can display in one line in the content table.

File: docs/source/getting-started/installation.rst (hunk 3 above)
Comment (Michaelvll) [reply docs/source/getting-started/installation.rst]: Oops, done. Thanks!

File: docs/source/reference/cloud-permissions.rst (hunk 5 above)
Comment (Michaelvll) [reply docs/source/reference/cloud-permissions.rst]: Oops, fixed. Thanks!

File: docs/source/reference/cloud-permissions.rst (hunk 7 above)
Comment (Michaelvll) [reply docs/source/reference/cloud-permissions.rst]: Good point! Changed.

File: docs/source/getting-started/installation.rst [hunk 10]
```diff
@@ -66,6 +68,8 @@ To get the **AWS access key** required by :code:`aws configure`, please go to th
 
 To use AWS IAM Identity Center (AWS SSO), see :ref:`here<aws-sso>` for instructions.
 
+[**Optional**] To create a new AWS user with minimal permissions for SkyPilot, see :ref:`AWS User Creation <cloud-permissions-aws>`.
```
Comment (concretevitamin) [inline docs/source/getting-started/installation.rst:71]: Nit: Can we `[**Optional**]` -> `**Optional**:`? Looks better to me. Same for GCP below?

File: docs/source/index.rst [hunk 11]
```diff
@@ -96,14 +96,12 @@ Documentation
    :maxdepth: 1
    :caption: User Guides
 
-   reference/local/index
    examples/iterative-dev-project
    reference/interactive-nodes
-   reference/cloud-administration
-   reference/quota
-   reference/logging
+   reference/clouds-setup/index
```
Comment (concretevitamin) [inline docs/source/index.rst:101]: In previous comment I was more thinking of adding a section parallel to User Guides:
```
.. toctree::
   :maxdepth: 1
   :caption: Cloud Administration and Usage
```
to hold these 3 pages. This way, an "administrator" will easily see that "oh they have docs for me".

Wdyt?

File: docs/source/reference/clouds-setup/cloud-permissions.rst [hunk 12]
```diff
@@ -127,9 +127,9 @@ AWS accounts can be attached with a policy that limits the permissions of the ac
 
 With the steps above you are almost ready to have the users in your organization to use SkyPilot with the minimal permissions.
 
-**One more thing** to do is to create a single service account "skypilot-v1" for all the users in your organization. There are two ways to accomplish this:
+**One more thing** to do is to create a single iam role ``skypilot-v1`` for all users in your organization. There are two ways to accomplish this:
```
Comment (concretevitamin) [inline docs/source/reference/clouds-setup/cloud-permissions.rst:130]: s/iam/IAM everywhere?

File: docs/source/reference/local/index.rst [hunk 13]
```diff
@@ -3,6 +3,10 @@
 SkyPilot On-prem
 ===================================
 
+.. note::
+
+   This feature has been **DEPRECATED**, and under the process of re-implementing. Please stay tuned for updates.
```
Comment (concretevitamin) [inline docs/source/reference/local/index.rst:8]:
```suggestion
   This feature has been **DEPRECATED**, and is under the process of re-implementation. Please stay tuned for updates.
```

File: docs/source/getting-started/installation.rst (hunk 10 above)
Comment (Michaelvll) [reply docs/source/getting-started/installation.rst]: Sounds good. Updated.

File: docs/source/index.rst (hunk 11 above)
Comment (Michaelvll) [reply docs/source/index.rst]: That also makes sense to me. Moved. Thanks!

File: docs/source/reference/clouds-setup/cloud-login.rst [hunk 14]
```diff
@@ -1,7 +1,7 @@
-.. _cloud-administration:
+.. _cloud-login:
 
-Cloud Administration
-=============================
+Using Cloud-Specific Login
```
[the remark this answers is not in the document]
Comment (Michaelvll) [reply docs/source/reference/clouds-setup/cloud-login.rst]: I like the "Authentication" better. Changed it and removed the L7-8 as the parent index has been removed.

File: docs/source/reference/clouds-setup/cloud-permissions.rst (hunk 12 above)
Comment (Michaelvll) [reply docs/source/reference/clouds-setup/cloud-permissions.rst]: Good catch! Done. Thanks!

File: docs/source/reference/clouds-setup/index.rst [hunk 15]
```diff
@@ -0,0 +1,14 @@
+.. _cloud-setup:
+
+Advanced Cloud Setup
```
[the remark this answers is not in the document]
Comment (Michaelvll) [reply docs/source/reference/clouds-setup/index.rst]: Removed index. Thanks!

File: docs/source/reference/local/index.rst (hunk 13 above)
Comment (Michaelvll) [reply docs/source/reference/local/index.rst]: Oops. Good catch! Fixed. Thanks!

**Review by concretevitamin — APPROVED** [review]
Thanks a lot for adding these docs @Michaelvll!

**Pull request merged**